inc/saiport.h

```diff
@@ -2400,6 +2447,40 @@ typedef enum _sai_port_attr_t
      */
     SAI_PORT_ATTR_DATAPATH_ENABLE,
 
+    /**
+     * @brief Read ethernet copper cable pair status.
+     *
+     * Returns pair states sequentially from list index 0 to n (n = number of pairs - 1)
+     * value of n depends on number of pairs in twisted ethernet copper cable
+     * Cable diagnostics triggers ECD once cable pair states is requested, it is synchronous call.
+     *
+     * @type sai_s32_list_t sai_port_cable_pair_state_t
+     * @flags READ_ONLY
+     */
+    SAI_PORT_ATTR_CABLE_PAIR_STATE,
+
+    /**
+     * @brief Get ethernet copper cable pair length (unit meter)
+     *
+     * Returns ethernet cable pair length sequentially from list index 0 to n (n = number of pairs - 1)
+     * value of n depends on number of pairs in twisted ethernet copper cable
+     * Cable diagnostics runs once cable pair length is requested, it is synchronous call.
+     *
+     * @type sai_s32_list_t
+     * @flags READ_ONLY
+     */
+    SAI_PORT_ATTR_CABLE_PAIR_LENGTH,
```

Comment:
@prafull-brcm is it the length of the DAC/copper cable? If not, what exactly is this length indicate?

Reply:
@prgeor It indicates copper cable length for example twisted wire pairs, coaxial cables with tolerance of +-1meter